src/main/java/com/couchbase/lite/replicator/PullerInternal.java

```diff
@@ -379,6 +378,8 @@ public void onCompletion(Response httpResponse, Object result, Throwable e) {
                             --httpConnectionCount;
                             // Start another task if there are still revisions waiting to be pulled:
                             pullRemoteRevisions();
+
```

Comment:
Can you remove the cancellable in here so you don't need the CancellableManager and the reverseCancellables map? 

For example:
```Java
cancellables.values().remove(this);
```

Reply:
I don't have a big picture about threading (what thread will this onCompletion get called). Hope there is no race issue here.